core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -4826,5 +4826,18 @@ private static void _setJenkinsJVM(boolean jenkinsJVM) {
             JenkinsJVM.setJenkinsJVM(jenkinsJVM);
         }
     }
+    /**
+     * This handles the situation when Parameterized build checkbox is checked 
+     * but no parameters are selected. User will be redirected to an error page
+     * with proper error message.
+     * @param jsonProperties
+     * @throws FormException 
+     */
+    public static void checkForEmptyParameters(JSONObject jsonProperties) throws FormException{
```

Comment:
I do not see this method as a part of public API (especially in the Jenkins class). There are hardcoded values for jobs only. Would be better to just inline this method to the code or to move it to a private method in the Job class